code/controllers/DMSDocumentCartController.php

```diff
@@ -0,0 +1,153 @@
+        '$Action//$ID' => 'handleAction',
+    );
+
+    private static $allowed_actions = array(
+        'DocumentCartForm',
+        'add',
+        'deduct',
+        'remove',
+    );
+
+    /**
+     * See {@link DMSDocumentCart::getItems()}
+     *
+     * @return ArrayList
+     */
+    public function Items()
+    {
+        return ArrayList::create($this->getCart()->getItems());
+    }
+
+    /**
+     * Prepares receiver info for the template.
+     * Additionally it uses Zend_Locale to retrieve the localised spelling of the Country
+     *
+     * @return ArrayData
+     */
+    public function getReceiverInfo()
+    {
+        $receiverInfo = $this->getCart()->getReceiverInfo();
+
+        if (isset($receiverInfo['DeliveryAddressCountry']) && $receiverInfo['DeliveryAddressCountry']) {
+            $source = Zend_Locale::getTranslationList('territory', $receiverInfo['DeliveryAddressCountry'], 2);
+            $receiverInfo['DeliveryAddressCountryLiteral'] = $source[$receiverInfo['DeliveryAddressCountry']];
+        }
+
+        if (!empty($receiverInfo)) {
+            $result = ArrayData::create($receiverInfo);
+        } else {
+            $result = ArrayData::create(array('Result' => 'no data'));
+        }
+
+        return $result;
+    }
+
+    /**
+     * See DMSDocumentCart::isCartEmpty()
+     *
+     * @return bool
+     */
+    public function getIsCartEmpty()
+    {
+        return $this->getCart()->isCartEmpty();
+    }
+
+    /**
+     * Add quantity to an item that exists in {@link DMSDocumentCart}.
+     * If the item does nt exist, try to add a new item of the particular
+     * class given the URL parameters available.
+     *
+     * @param SS_HTTPRequest $request
+     */
+    public function add(SS_HTTPRequest $request)
+    {
+
+        $quantity = ($request->requestVar('quantity')) ? intval($request->requestVar('quantity')) : 1;
+        $documentId = (int)$request->param('ID');
+        /** @var DMSDocument|DMSDocumentCartExtension $doc */
+        if ($doc = DMSDocument::get()->byID($documentId)) {
+            if ($doc->AllowedInCart && $doc->canView()) {
+                if ($this->getCart()->getItem($documentId)) {
+                    $this->getCart()->updateItemQuantity($documentId, $quantity, DMSDocumentCart::ITEM_QUANTITY_ADD);
+                } else {
+                    $requestItem = DMSRequestItem::create()->setDocument($doc)->setQuantity($quantity);
+                    $this->getCart()->addItem($requestItem);
+                }
+                $backURL = $request->getVar('BackURL');
+                // make sure that backURL is a relative path (starts with /)
+                if (isset($backURL) && preg_match('/^\//', $backURL)) {
+                    $this->getCart()->setBackUrl($backURL);
+                }
+            }
+        }
+
+        if ($request->isAjax()) {
+            $this->response->addHeader('Content-Type', 'text/plain');
+            echo true;
```

Comment:
Updated Content-Type to application/json and used jsin_encode to return the result.